smartconfig/configuration.py

```diff
@@ -0,0 +1,173 @@
+import inspect
+from itertools import chain
+from typing import Any, Dict, List, Optional, Tuple
+
+from smartconfig.exceptions import ConfigurationKeyError, DuplicateConfiguration, InvalidOperation
+from smartconfig.parser import YamlLikeParser
+from smartconfig.register import _register
+from smartconfig.typehints import _EntryMapping, _FilePath
+
+
+def load_config_file(path: _FilePath) -> None:
```

Comment:
> As I stated before, it should be possible to write new values in a non-destructive manner. Furthermore, I don't foresee why it'd need to overwrite existing values.

I don't understand why (and how) we should write new values without replacing the previous ones, at least with the current setup. We don't have any needs for the old values, we can just get rid of them.

Reply:
> Furthermore, I want to reiterate that this shouldn't be necessary for the server bootstrapping. In fact, I believe it'd be more cumbersome to have multiple files. 

I'd also like to point you out to [this](https://discord.com/channels/267624335836053506/792787094468689940/792788629608595476) message in the server outlining my current thoughts about the configuration, so we can share opinions on that :grin:

Reply:
The bootstrapper only needs to write to the config the IDs of the Discord objects it creates. Neither default values nor IDs for existing Discord objects will be written. If the config already contains a given ID and it points to something valid, then no new Discord object needs to be created. Therefore, that value doesn't need to be overwritten. However, if an ID is invalid, then we can discuss the behaviour later e.g. prompting the user or overwriting invalid IDs without any prompt.

My point is that all of this can be done with a single file. The implementation won't require one file for user overrides and another auto-generated one from the bootstrapper. Therefore, facilities for supporting multiple file loads are unnecessary,

Reply:
Alright, if I understand correctly, you want to put all the discord IDs under the same YAML-category, and not write it again in all the entries that use it?

If so, yeah, I don't think we should have many configuration files, that's a good idea.